src/Arcus.EventGrid.Tests.Integration/Publishing/CloudEventPublishing.cs

```diff
@@ -0,0 +1,196 @@
+﻿using System;
+using System.Net.Mime;
+using System.Threading.Tasks;
+using Arcus.EventGrid.Contracts;
+using Arcus.EventGrid.Parsers;
+using Arcus.EventGrid.Publishing.Interfaces;
+using Arcus.EventGrid.Tests.Core.Events.Data;
+using Arcus.EventGrid.Tests.Integration.Fixture;
+using CloudNative.CloudEvents;
+using Newtonsoft.Json;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Arcus.EventGrid.Tests.Integration.Publishing
+{
+    [Trait(name: "Category", value: "Integration")]
+    public class CloudEventPublishing : IAsyncLifetime
+    {
+        private readonly ITestOutputHelper _testOutput;
+        private EventGridTopicEndpoint _endpoint;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CloudEventPublishing"/> class.
+        /// </summary>
+        public CloudEventPublishing(ITestOutputHelper testOutput)
+        {
+            _testOutput = testOutput;
+        }
+
+        /// <summary>
+        /// Called immediately after the class has been created, before it is used.
+        /// </summary>
+        public async Task InitializeAsync()
+        {
+            _endpoint = await EventGridTopicEndpoint.CreateForCloudEventAsync(_testOutput);
```

Comment:
See EventGridTopicEndpoint, I'd argue this class does 2 things:
- Provide an endpoint to create publisher for (which we do for every test via BuildPublisher, while we could only do it once? 🤔 )
- Provide host to receive events on

Reply:
Ah, yes, we could maybe move the creation logic of the event publisher to somehwere else.

Reply:
LGTM